hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/WALCellCodec.java

```diff
@@ -302,18 +367,28 @@ protected Cell parseCell() throws IOException {
         compression.getDictionary(CompressionContext.DictionaryIndex.QUALIFIER));
       pos += elemLen;
 
-      // timestamp, type and value
-      int tsTypeValLen = length - pos;
+      // timestamp
+      long ts = StreamUtils.readLong(in);
+      pos = Bytes.putLong(backingArray, pos, ts);
+      // type and value
+      int typeValLen = length - pos;
       if (tagsLength > 0) {
-        tsTypeValLen = tsTypeValLen - tagsLength - KeyValue.TAGS_LENGTH_SIZE;
+        typeValLen = typeValLen - tagsLength - KeyValue.TAGS_LENGTH_SIZE;
+      }
+      byte type = (byte)in.read();
+      pos = Bytes.putByte(backingArray, pos, type);
+      int valLen = typeValLen - 1;
+      if (compression.hasValueCompression()) {
```

Comment:
Same here on avoiding a branch, should we do that?

Reply:
Yes. 

Also applies to the check for if tag compression is enabled as well.